src/modules/tokens/services/token.compute.service.ts

```diff
@@ -275,19 +275,80 @@ export class TokenComputeService implements ITokenComputeService {
         const previous24hPriceBN = new BigNumber(previous24hPrice);
 
         if (previous24hPriceBN.isZero()) {
-            if (currentPriceBN.isZero()) {
-                return '0';
-            }
+            return '0';
+        }
+
+        return currentPriceBN.dividedBy(previous24hPrice).toFixed();
+    }
+
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'token',
+        remoteTtl: CacheTtlInfo.Token.remoteTtl,
+        localTtl: CacheTtlInfo.Token.localTtl,
+    })
+    async tokenVolumeChange24h(tokenID: string): Promise<string> {
+        return await this.computeVolumeChange24h(tokenID);
+    }
+
+    async computeVolumeChange24h(tokenID: string): Promise<string> {
+        const min_24h_volume = 10000;
 
-            return '10000';
+        const [currentVolume, previous24hVolume] = await Promise.all([
+            this.tokenVolumeUSD24h(tokenID),
+            this.tokenPrevious24hVolumeUSD(tokenID),
+        ]);
+
+        const currentVolumeBN = new BigNumber(currentVolume);
+        const previous24hVolumeBN = new BigNumber(previous24hVolume);
+
+        if (currentVolumeBN.isZero()) {
+            return '0';
         }
 
-        const difference = currentPriceBN.minus(previous24hPriceBN);
+        const maxPrevious24hVolume = BigNumber.maximum(
+            previous24hVolumeBN,
+            min_24h_volume,
+        );
+
+        return currentVolumeBN.dividedBy(maxPrevious24hVolume).toFixed();
+    }
 
-        return difference
-            .dividedBy(previous24hPriceBN)
-            .multipliedBy(100)
-            .toFixed();
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'token',
+        remoteTtl: CacheTtlInfo.Token.remoteTtl,
+        localTtl: CacheTtlInfo.Token.localTtl,
+    })
+    async tokenTradeChange24h(tokenID: string): Promise<string> {
+        return await this.computeTradeChange24h(tokenID);
+    }
+
+    async computeTradeChange24h(tokenID: string): Promise<string> {
```

Comment:
I think you can return a number from these methods

Reply:
done